JavaFXUserInterface/src/main/java/SystemTypeInfoController.java

```diff
@@ -60,32 +62,111 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
-        thresholdAdvisory.setText(new DecimalFormat("#.##").format(assetType.getValue_advisory()));
-        thresholdCaution.setText(new DecimalFormat("#.##").format(assetType.getValue_caution()));
-        thresholdWarning.setText(new DecimalFormat("#.##").format(assetType.getValue_warning()));
-        thresholdFailed.setText(new DecimalFormat("#.##").format(assetType.getValue_failed()));
+        this.originalAssetType = new SystemTypeList(assetType);
+        systemTypeName.setText(assetType.getAssetType().getName());
+        systemTypeDesc.setText(assetType.getAssetType().getDescription());
+        try { thresholdOK.setText(new DecimalFormat("#.##").format(Double.parseDouble(assetType.getValueOk()))); } catch (NumberFormatException e){ }
+        try { thresholdAdvisory.setText(new DecimalFormat("#.##").format(Double.parseDouble(assetType.getValueAdvisory()))); } catch (NumberFormatException e){ }
+        try { thresholdCaution.setText(new DecimalFormat("#.##").format(Double.parseDouble(assetType.getValueCaution()))); } catch (NumberFormatException e){ }
+        try { thresholdWarning.setText(new DecimalFormat("#.##").format(Double.parseDouble(assetType.getValueWarning()))); } catch (NumberFormatException e){ }
+        try { thresholdFailed.setText(new DecimalFormat("#.##").format(Double.parseDouble(assetType.getValueFailed()))); } catch (NumberFormatException e){ }
+
+
     }
 
     /**
      * Attaches events to elements in the scene.
      *
-     * @author Najim, Paul
+     * @author Najim
+     *
+     * Edit: added all the text proprety listeners and text formaters for all the fields
+     *
+     * @author Paul
      */
     public void attachEvents() {
         // Change scenes to Systems.fxml
         systemMenuBtn.setOnMouseClicked(mouseEvent -> uiUtilities.changeScene(mouseEvent, "/Systems"));
         //Attach link to systemTypeMenuBtn to go to SystemTypeList.fxml
         systemTypeMenuBtn.setOnMouseClicked(mouseEvent -> uiUtilities.changeScene(mouseEvent, "/SystemTypeList"));
-        //Attach link to infoEditBtn to go to SystemTypeEdit.fxml
-        infoEditBtn.setOnMouseClicked(mouseEvent -> uiUtilities.changeScene(mouseEvent, "/SystemTypeEdit", assetType));
-        //Attach the asset type delete function
         infoDeleteBtn.setOnMouseClicked(this::deleteDialog);
+
+        infoSaveBtn.setDisable(true);
+        infoSaveBtn.setOnMouseClicked(mouseEvent -> {
+            assetTypeDAO.updateAssetType(assetType.toAssetType());
+            uiUtilities.changeScene(mouseEvent, "/SystemTypeList");
+        });
+
+        systemTypeName.textProperty().addListener((obs, oldText, newText) -> {
+            if( handleTextChange(newText,originalAssetType.getName()))
+                assetType.getAssetType().setName(newText);
+        });
+        systemTypeDesc.textProperty().addListener((obs, oldText, newText) -> {
+            if( handleTextChange(newText,originalAssetType.getDescription()))
```

Comment:
When calling the handleTextChange for description, it seems like the application is trying to parse a double. Maybe there is something wrong going in the if conditions of the method.
![SetDescription issue](https://user-images.githubusercontent.com/47158909/105775260-86d55c80-5f34-11eb-88d0-3eaa6607e328.PNG)